### ISSUE TITLE
Anyone have load_and_authorize_resource working through a has_one association?

### ISSUE THREAD

**Issue by softwaregravy** [CanCanCommunity/cancancan#622]
tldr; a `has_one` relation doesn't support `new` so I don't think you can use `load_and_authorize_resource through: x`. 

I have a `has_one` relationship on the model and trying to do a `load_and_authorize_resource :through`, but it doesn't seem to work because you can't call `new` on the relation. I've tried using the `:through_relation` option, but this is the same problem. Seem to only be valid for a `has_many` relationship. 

Call to [new](https://github.com/CanCanCommunity/cancancan/blob/3.0.2/lib/cancan/controller_resource_builder.rb#L8)

Has anyone had this working? 

Sorry I don't have time to get a separate project set up I can share with this demo'd

I guess if we agree this is true, I can add some notes to the docs just to call this out. 

### System configuration
**Rails version**: 5.2.4

**Ruby version**:

**CanCanCan version** 3.0.2
